frozendict/tests.py

```diff
@@ -0,0 +1,131 @@
+from unittest import TestCase
+
+from frozendict import frozendict, FrozenOrderedDict
+
+
+class BaseTestCase(TestCase):
+    def assertMappingEqual(self, m1, m2):
+        # `assertDictEqual` will make a failing type check when comparing a `dict`
+        # with a `frozendict`. Instead, we'll compare the item sets:
+        self.assertSetEqual(set(m1.items()), set(m2.items()))
+
+    def assertRaisesAssignmentError(self, *args, **kwargs):
+        self.assertRaisesRegexp(
+            TypeError,
+            r'object does not support item assignment',
+            *args, **kwargs
+        )
+
+
+def setitem(dct, key, value):
+    dct[key] = value
+
+
+class TestFrozenDict(BaseTestCase):
+    def test_assignment_fails(self):
+        self.assertRaisesAssignmentError(setitem, frozendict(), 'key', 'value')
```

Comment:
```python
with self.assertRaises(TypeError):
    frozendict.key = value
```
but `assertRaises` needs to be backported. I suggest that you copy B2 CLI [TestBase](https://github.com/Backblaze/B2_Command_Line_Tool/blob/4eee3ab58e80c087b81b5c748f481a902820478e/test/test_base.py#L16).

Reply:
Likewise not a problem in 2020 if dropping 2.6.